pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,264 @@
+	createTableStmt := "CREATE Table IF NOT EXISTS " + tableName +
+		"(`key` varchar(255) NOT NULL ,`value` BLOB, PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	_, err = tx.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        p.db,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return storage.ErrKeyRequired
+	}
+
+	//nolint: gosec create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, storage.ErrKeyRequired
+	}
+
+	var res result
+	// nolint: gosec select query to fetch the record by key
+	err := s.db.QueryRow("SELECT * FROM "+s.tableName+" "+
+		" WHERE `key` = ?", k).Scan(&res.key, &res.value)
+	if err != nil {
+		if strings.Contains(err.Error(), sqlDBNotFound) {
+			return nil, storage.ErrDataNotFound
+		}
+
+		return nil, err
+	}
+
+	return res.value, nil
+}
+
+// Delete will delete record with k key
+func (s *sqlDBStore) Delete(k string) error {
+	if k == "" {
+		return storage.ErrKeyRequired
+	}
+	// nolint: gosec delete query to delete the record by key
+	_, err := s.db.Exec("DELETE FROM "+s.tableName+" WHERE `key`= ?", k)
+
+	if err != nil {
+		return fmt.Errorf("failed to delete row %w", err)
+	}
+
+	return nil
+}
+
+type sqlDBResultsIterator struct {
+	store      *sqlDBStore
+	resultRows *sql.Rows
+	err        error
+}
+
+func (s *sqlDBStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	endKey = strings.ReplaceAll(endKey, storage.EndKeySuffix, "~")
+	//nolint:gosec sub query to fetch the all the keys that have start and end key reference, simulating range behavior.
+	queryStmt := "SELECT * FROM " + s.tableName + " WHERE `key` >= ?  AND `key` < ?"
```

Comment:
I have checked it. You need to add `ORDER BY key` as well. Also, I would suggest using `BETWEEN` operator. 

```
SELECT * FROM test 
WHERE `key` BETWEEN 'abc_' AND 'mno_123~' ORDER BY `key`
```

Reply:
As discussed offline, will stick to operand query for now. Using Between always include the endkey value which we dont want